wptherml/tests/test_exciton.py

```diff
@@ -0,0 +1,117 @@
+"""
+Unit and regression test for the wpspec package.
+"""
+
+# Import package, test suite, and other packages as needed
+import wptherml
+import numpy as np
+import pytest
+import sys
+
+
+""" perform a series of tests that would apply to the simple
+    case of a r = 100e-9 m spehere made of glass with ri = 1.5+0j
+    in air with ri = 1.0+0j
+"""
+sf = wptherml.SpectrumFactory()
+args = {
+'exciton_energy': 1.5,
+'aggregate_shape': (2,1,1),
+'displacement_vector' : [10000, 0, 0],  
+'transition_dipole_moment' : np.array([0, 0, 0.5]),
+'refractive_index' : 1.0,
+}
+
+exciton_test = sf.spectrum_factory('Frenkel', args)
+
+""" Define a second test instance that will
+    provide a simple case to test the dynamics... simple meaning
+    no coupling, so the transition_dipole_moment is set to zero
+    so that the V terms are 0 and so any localized exciton state
+    (i.e. c_vector = [1, 0] or c_vector = [0, 1] will be an eigenstate)
+    of the exciton Hamiltonian
+"""
+dynamics_args = {
+'exciton_energy': 1,
+'aggregate_shape': (2,1,1),
+'displacement_vector' : [1000000, 0, 0],  
+'transition_dipole_moment' : np.array([0, 0, 0.0]),
+'refractive_index' : 1.0,
+}
+
+dynamics_test = sf.spectrum_factory("Frenkel", dynamics_args)
+
+def test_compute_H0_element():
+
+    _H_11 = exciton_test._compute_H0_element(1, 1)
+    _H_12 = exciton_test._compute_H0_element(1, 2)
+
+    _expected_H_11 = exciton_test.exciton_energy
+    _expected_H_12 = 0.
+
+    assert np.isclose(_H_11, _expected_H_11, 1e-5)
+    assert np.isclose(_H_12, _expected_H_12, 1e-5)
+
+
+def test_compute_dipole_dipole_coupling():
```

Comment:
@lanetolley This unit test needs to be improved, it only tests a trivial case for when the transition dipole moment is zero and the monomers are separated by a very large distance.  

You should write a test for several different cases that will make sure that the calculation works under a variety of displacements and for the different instances of a transition dipole moment that might arise.

For example, can you have a TDM with non-zero x, y, and z components and then have unit tests for when:

a. the monomers are displaced along x only but by a small amount
b. the monomers are displaced along y only but by a small amount
c. the monomers are displaced along z only but by a small amount
d. the monomers are displaced along x, y, and z but by a small amount